### PR TITLE
[improve](serde) support json string format with escaped charactors

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
before this pr
if we use some escape from nested types like array, we  can not make this string cast to json
Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_string_serde.h [hunk 1]
```diff
@@ -73,16 +73,52 @@ class DataTypeStringSerDeBase : public DataTypeSerDe {
...
             bw.write('"');
         }
-
-        const auto& value = assert_cast<const ColumnType&>(*ptr).get_data_at(row_num);
-        bw.write(value.data, value.size);
+        if constexpr (std::is_same_v<ColumnType, ColumnString>) {
+            if (options.escape_char != 0) {
+                // we should make deal with some special characters in json str if we have escape_char
+                StringRef str_ref = value;
+                for (char it : str_ref) {
```
Comment (eldenmoon) [inline be/src/vec/data_types/serde/data_type_string_serde.h:85]: put this into a function may enhance readability

**Comment by qidaye** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_string_serde.h (hunk 1 above)
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_string_serde.h]: done!

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
this pr is no need to pick for 2.0 , because it has bavior change , and it relays on https://github.com/apache/doris/pull/34764

**Pull request merged**